### PR TITLE
feat: HUBSA-2346 use profile for getting maven artifacts

### PR DESCRIPTION
**Pull request by rbrisuda**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -118,16 +129,18 @@ runs:
     - name: Download Maven S3 repository extension
       shell: bash
       run: |
+        TMP_WAGON_PATH="/tmp/s3-wagon-${{ inputs.maven-wagon-version }}-jar-with-dependencies.jar"
+        aws s3 cp ${{ env.MAVEN_WAGON_PATH }} $TMP_WAGON_PATH --profile ${{ steps.configure-maven-aws-profile.outputs.aws-profile }}
```
Comment (rbrisuda) [inline action.yml]: In this way I workarounded that I do not need to have aws credentials in "sudo" context so we can use that profile outside of action. So I will only copy it with sudo on place which needs sudo access.

File: action.yml [hunk 2]
```diff
@@ -59,18 +59,17 @@ outputs:
     value: ${{ steps.maven_changed_modules.outputs.changed_modules }}
   maven-aws-profile:
     description: "Aws profile used for accessing maven repositories."
-    value: ${{ steps.configure-maven-aws-profile.outputs.aws-profile }}
+    value: ${{ inputs.maven-aws-profile-name }}
```
Comment (mmichna) [inline action.yml:62]: in **run-maven-gh-action** you use `inputs.maven-aws-profile` - maybe is good use same input name

**Review by mmichna — APPROVED** [review]

File: action.yml [hunk 3]
```diff
@@ -9,20 +9,24 @@ inputs:
...
+  maven-aws-role-to-assume:
+    description: "Role to assume for accessing maven artifacts S3 repositories"
     required: true
-  account-id:
-    description: "Account id where maven artifacts S3 repositories exist."
+  maven-artifacts-account-id:
+    description: "Account id where maven artifacts S3 repositories exist"
     required: true
+  maven-aws-profile-name:
+    description: "Profile name for aws profile to be created."
```
Comment (mmichna) [inline action.yml]: remove "."

File: action.yml [hunk 4]
```diff
@@ -9,20 +9,24 @@ inputs:
...
-    description: "Secret key for credentials for accessing maven artifacts S3 repositories"
+  maven-aws-role-to-assume:
+    description: "Role to assume for accessing maven artifacts S3 repositories"
     required: true
-  account-id:
-    description: "Account id where maven artifacts S3 repositories exist."
+  maven-artifacts-account-id:
+    description: "Account id where maven artifacts S3 repositories exist"
     required: true
+  maven-aws-profile-name:
```
Comment (mmichna) [inline action.yml:21]: rename to `maven-aws-profile` - same like in run GH action

**Review by mmichna — DISMISSED** [review]

File: action.yml (hunk 2 above)
Comment (rbrisuda) [reply action.yml:62]: I used: aws-profile-name on places where profile does not exists so it is your input, what profile under which name should be created. Without "name", it is allready created profile to be used.
If it does not worth to distinguish between that two case I can remove "name" from all of them.

Comment (mmichna) [reply action.yml:62]: @MartinMitro ??? ^

**Review by MartinMitro — APPROVED** [review]

**Review by mmichna — APPROVED** [review]

**Review by jcajkovic — APPROVED** [review]

**Pull request merged**